Code/GraphMol/SubstructLibrary/SubstructLibrary.cpp

```diff
@@ -86,6 +89,64 @@ struct Bits {
     }
   }
 
+  // FIX complete this
+  Bits(const FPHolderBase *fingerprints, const ExtendedQueryMol &xqm,
+       const SubstructMatchParameters &ssparams)
+      : fps(fingerprints), params(ssparams) {
+    if (fps) {
+      const auto *tph = dynamic_cast<const TautomerPatternHolder *>(fps);
+      const auto *ph = dynamic_cast<const PatternHolder *>(fps);
+      if (std::holds_alternative<ExtendedQueryMol::RWMol_T>(xqm.xqmol)) {
+        queryBits = fps->makeFingerprint(
+            *std::get<ExtendedQueryMol::RWMol_T>(xqm.xqmol));
+      } else if (std::holds_alternative<ExtendedQueryMol::MolBundle_T>(
+                     xqm.xqmol)) {
+        auto &bndl = std::get<ExtendedQueryMol::MolBundle_T>(xqm.xqmol);
+        auto tqb = new ExplicitBitVect(ph->getNumBits());
```

Comment:
I would find this less confusing as

```
queryBits = new ExplicitBitVect(ph->getNumBits());
...
   *queryBits &= *tfp;
```

Reply:
`queryBits` is const, so we can't do this.

Reply:
Ah, I wonder if that's necessary.